Improve flake

TornaxO7 opened the pull request:
this makes the flake.nix file a bit more readable in my opinion :)

The changes are:
- Replacing `flake-utils` with [`flake-parts`] because [`flake-parts`] is "syntax-friendlier"
- making use of [`oxalica/rust-overlay`] in order to load the given rust-toolchain
- added weekly CI to keep `flake.lock` up to date
- I'd also volunteer to maintain the `flake.nix`
- updated `flake.lock` with `nix flake update`

@Toqozz 

[`flake-parts`]: https://flake.parts/
[`oxalica/rust-overlay`]: https://github.com/oxalica/rust-overlay

TornaxO7 commented:
Hm.. actually... I'm rethinking this a bit.

Toqozz commented:
Thanks!

I'd love for any of the other nix contributors to have a look at this since I don't use nix personally.  @SignalWalker 

If nobody gets to this within a week I'll try to take a look myself.  Could you briefly explain the improvements?

TornaxO7 commented:
> Could you briefly explain the improvements?

- (Subjective improvement): Switching from [`flake-utils`] to [`flake-parts`] which has a cleaner structure for multiple systems
- updated the `flake.lock` with `nix flake update` (you can't currently run `cargo check` because it's version is too old)
- added [`rust-overlay`] so the nix-dev-shells are loading the tools from the given `rust-toolchain.toml`

[`flake-utils`]: https://github.com/numtide/flake-utils
[`flake-parts`]: https://flake.parts/
[`rust-overlay`]: https://github.com/oxalica/rust-overlay

SignalWalker reviewed (approved):
Looks good to me, though you might want to keep Alejandra or get some other formatter.

TornaxO7 commented:
> though you might want to keep Alejandra or get some other formatter.

Fixed, I added the [`nixpkgs-fmt`](https://search.nixos.org/packages?channel=unstable&show=nixpkgs-fmt&from=0&size=50&sort=relevance&type=packages&query=nixpkgs-fmt) :)

Toqozz commented:
Awesome, thanks so much you guys.

TornaxO7 commented:
![peepoAwesome](https://github.com/Toqozz/wired-notify/assets/50843046/0c97428d-1fd3-4921-a68d-143b71a72bf4)
